File: 9_palindrome_number/palindrome_number.py

class Solution:
    def isPalindrome(self, x: int) -> bool:
        _x, y = x, 0
        if x < 0:
            return False
        while x:
            mantissa = x % 10
            x //= 10
            y = y * 10 + mantissa
        return _x == y

File: 9_palindrome_number/palindrome_number.java
import java.util.Stack;

class Solution {
    // 借助栈判断回文字符串
    public boolean isPalindrome(int x) {
        if (x < 0) return false;
        String a = String.valueOf(x);
        Stack<String> stack = new Stack<String>();
        for (int i = 0; i < a.length() ; i++) {
            stack.push(a.substring(i, i + 1));
        }
        for (int i = 0; i < a.length() ; i++) {
            String tmp = stack.pop();
            if (!tmp.equals(a.substring(i, i + 1))) return false;
        }
        return true;
    }
}